_posts/2019-11-08-method-references-the-unpolished-gem-of-java-8.markdown

```diff
@@ -0,0 +1,299 @@
+---
+layout: post
+title:  "Method References - the unpolished gem of Java 8"
+date:   2019-11-08 17:16:00 +0100
+categories: Java, Language
+---
+
+If you were to pick your favorite year from the entire history of mankind, which year would it be?
+Most people would probably say 1932, when [the Great Emu War](https://www.nationalgeographic.com.au/history/the-great-australian-emu-war.aspx) was fought in Australia.
+This one is definitely in my top 10; however it falls behind 2014 - the year when Java 8 has been released.
+
+Among other things, this glorious language update added some functional programming capabilities to the language.
+Whether or not you consider this a good idea, [Lambda expressions](https://www.jcp.org/en/jsr/detail?id=335) are now an unavoidable reality of (almost) any Java project.
+
+Therefore, we should look for ways to make the code that uses them as clean and concise as possible.
+
+## Lambda expressions and Method References
+
+[Method References](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.13)—also introduced in Java 8—provide a different syntax for Lambda expressions.
+They may be used with any type of method: instance, static or a constructor.
+Although often overlooked, I believe they have a power to greatly improve our code.
+
+I will not talk about how to use Method References here.
+This has been explained in great detail by others already, for example in [this tutorial from Oracle](https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html).
+My goal is to show the advantages of using them and outline ideas on how they could be improved in the future Java releases.
+
+## Why Method References are great
+
+For many, method references are just a bit of synthetic sugar sprinkled about in the language.
```

Comment:
synthetic => syntactic sugar ? or is it intentional?

Reply:
Not intentional. Good catch, thanks! (:
Fixed here: https://github.com/Jezorko/Jezorko.github.io/pull/1/commits/478efeefcbe646c917d5163395c5c22c1e52861f